### PR TITLE
Update to support bundling

### PR DESCRIPTION
**Pull request by geigerj**
Provides access to bundling through ApiCallable. Replaces
ApiCallDefaults with CallSettings, which corresponds to a change
in the code generation to pass bundling settings through to
ApiCallable.

### CODE REVIEW

File: google/gax/api_callable.py [hunk 1]
```diff
@@ -262,63 +265,42 @@ class ApiCallable(object):
...
-                timeout)
+                self.settings.page_descriptor.request_page_token_field,
+                self.settings.page_descriptor.response_page_token_field,
+                self.settings.page_descriptor.resource_field,
+                self.settings.timeout)
         else:
-            the_func = _add_timeout_arg(the_func, timeout)
+            the_func = _add_timeout_arg(the_func, self.settings.timeout)
+            if self.settings.bundler and self.settings.bundle_descriptor:
+                the_func = _bundleable(
```
Comment (tbetbetbe) [inline google/gax/api_callable.py]: N.B: At this point, _bundleable is called with the_func which is still a callable([[request], response])
I refer to this in another comment.

File: test/test_api_callable.py [hunk 2]
```diff
@@ -108,82 +111,86 @@ def grpc_return_value(request, *dummy_args, **dummy_kwargs):
...
+    def test_bundling(self):
+        # pylint: disable=abstract-method, too-few-public-methods
+        class BundlingRequest(object):
+            def __init__(self, messages=None):
+                self.messages = messages
+
+        fake_grpc_func_descriptor = BundleDescriptor('messages', [])
+        bundler = bundling.Executor(BundleOptions(message_count_threshold=8))
+
+        def my_func(request, dummy_timeout):
```
Comment (tbetbetbe) [inline test/test_api_callable.py]: See the other comment - is there a need for dummy_timeout here ?

File: google/gax/api_callable.py (hunk 1 above)
Comment (geigerj) [inline google/gax/api_callable.py]: We start out with (if you'll forgive some haskell-ish notation):
the_func :: (RequestType, int) -> (ResponseType)
_add_timeout_arg :: ((type_1, ..., type_n, int) -> X), int) -> (type_1, ..., type_n) -> X

hence,
_add_timeout_arg(the_func) :: RequestType -> ResponseType

So I think in test_api_callable, my_func must take a dummy_timeout, or else we can't meaningfully apply _add_timeout_arg to it in ApiCallable. Does that sounds right?

Comment (tbetbetbe) [inline google/gax/api_callable.py]: that makes sense, LGTM

**Comment by tbetbetbe** [comment]
LGTM

**Pull request merged**